src/jit/codegenxarch.cpp

```diff
@@ -420,7 +420,13 @@ void CodeGen::instGen_Set_Reg_To_Imm(emitAttr size, regNumber reg, ssize_t imm,
     {
         if (genDataIndirAddrCanBeEncodedAsPCRelOffset(imm))
         {
-            getEmitter()->emitIns_R_AI(INS_lea, EA_PTR_DSP_RELOC, reg, imm);
+            emitAttr newSize = EA_PTR_DSP_RELOC;
+            if (EA_IS_BYREF(size))
+            {
+                newSize = EA_SET_FLG(newSize, EA_BYREF_FLG);
+            }
+
+            getEmitter()->emitIns_R_AI(INS_lea, newSize, reg, imm);
```

Comment:
Is there a reason why the `INS_mov` case below doesn't also need the BYREF code?

Reply:
The PCRel case is special as it creates a new EA out of thin air and so loses any caller contribution passed via `size`. So we need to merge the caller-supplied gcref part back in.

In the non PCRel case `size` is the value passed by the caller so no merging is needed.

Reply:
@BruceForstall does that make sense?

Reply:
Yep, makes sense